tests/utils.go

```diff
@@ -237,12 +237,29 @@ type ProcessFunc func(event *k8sv1.Event) (done bool)
 type ObjectEventWatcher struct {
 	object                 runtime.Object
 	timeout                *time.Duration
-	failOnWarnings         bool
 	resourceVersion        string
 	startType              startType
+	warningPolicy          WarningsPolicy
 	dontFailOnMissingEvent bool
 }
 
+type WarningsPolicy struct {
+	FailOnWarnings     bool
+	WarningsIgnoreList []string
+}
+
+func (wp *WarningsPolicy) shouldIgnoreWarning(event *k8sv1.Event) bool {
+	if event.Type == string(WarningEvent) {
+		for _, message := range wp.WarningsIgnoreList {
+			if message == event.Message {
```

Comment:
@oshoval I am not sure if we need to trim newlines and spaces from the message we receive from the watcher, but if it's redundant please ignore my comment.

Reply:
we don't need to, this is the exact string that happens, and its kept on event.Message